packages/web-config-server/src/apiV1/measureData.js

```diff
@@ -330,18 +330,42 @@ export default class extends DataAggregatingRouteHandler {
       entityCode: this.entity.code,
       isDataRegional: dataServices?.[0]?.isDataRegional,
     });
-    dhisApi.injectFetchDataSourceEntities(this.fetchDataSourceEntities);
-    const buildMeasure = async (measureId, ...args) => ({
-      [measureId]: await getMeasureBuilder(measureBuilder)(...args),
-    });
 
+    if (legacy) {
+      const {
+        data_builder: measureBuilder,
+        data_builder_config: measureBuilderConfig,
+      } = mapOverlay;
+      const { dataElementCode = 'value' } = measureBuilderConfig;
+      dhisApi.injectFetchDataSourceEntities(this.fetchDataSourceEntities);
+      const buildMeasure = async (measureCode, ...args) => ({
+        [measureCode]: await getMeasureBuilder(measureBuilder)(...args),
+      });
+
+      return buildMeasure(
+        code,
+        this.models,
+        this.aggregator,
+        dhisApi,
+        { ...this.query, dataElementCode },
+        { ...measureBuilderConfig, dataServices },
+        entity,
+        this.req,
+      );
+    }
+
+    // NON-LEGACY
+
+    const buildMeasure = async (measureCode, ...args) => ({
+      [measureCode]: await getMeasureBuilder('useReportServer')(...args),
+    });
     return buildMeasure(
-      id,
+      code,
       this.models,
       this.aggregator,
       dhisApi,
-      { ...this.query, dataElementCode },
-      { ...measureBuilderConfig, dataServices },
+      { ...this.query, dataElementCode: 'value' },
+      { reportCode: mapOverlay.report_code },
       entity,
       this.req,
     );
```

Comment:
We don't need to maintain consistency with the legacy way here - I think it'd be cleaner to just call into the report server builder directly (e.g. we're otherwise passing in two arguments that are unused, and I also can't see that the report server builder uses `dataElementCode` at all, etc.)

Something like the following might work, but needs testing:
```
    const builder = new ReportServerBuilder(
      this.req,
      this.models,
      { reportCode: mapOverlay.report_code },
      { ...this.query },
      entity,
    );
    return {
      [measureCode]: await builder.build(),
    };
```